runtime/parachains/src/inclusion.rs

```diff
@@ -427,12 +427,15 @@ impl<T: Config> Pallet<T> {
 	///
 	/// Both should be sorted ascending by core index, and the candidates should be a subset of
 	/// scheduled cores. If these conditions are not met, the execution of the function fails.
-	pub(crate) fn process_candidates(
+	pub(crate) fn process_candidates<GV, const SKIP_CHECKER_CHECKS: bool>(
```

Comment:
some docs on SKIP_CHECKER_CHECKS would be nice
also having this is a const generics seems like a premature optimization to me (increasing the wasm size), unless it provides some type safety guarantess which I don't see

Reply:
That's very good point, replacing it with an enum is probably better.

Reply:
Also `if !do_not_do_thing { }` is harder to understand than `do_thing { }` in my opinion.